### PR TITLE
Httpclient as factory

### PR DESCRIPTION
**Pull request by cmlizama**
the same httpClient function was used in all three controllers, so I pulled it out into its own service.

### CODE REVIEW

**Comment by cmlizama** [comment]
hey @jkap

File: app/openBox/openBox.js [hunk 1]
```diff
@@ -4,17 +4,10 @@ angular.module('bby-query-mixer.openBox').controller('openBoxCtrl', [
     '$scope',
     '$resource',
     'categoryConfig',
-    function ($scope, $resource, categoryConfig) {
+    'HttpClientService',
+    function ($scope, $resource, categoryConfig, HttpClientService) {
```
Comment (jkap) [inline app/openBox/openBox.js]: Do you still need to inject `$resource`?

File: app/productSearch/productSearch.js [hunk 2]
```diff
@@ -7,19 +7,12 @@ angular.module('bby-query-mixer.productSearch').controller('ProductSearchCtrl',
     '$resource',
     'showOptionsConfig',
     'attributeOptionsConfig',
-    function ($scope, categoryConfig, $http, $resource, showOptionsConfig, attributeOptionsConfig) {
+    'HttpClientService',
+    function ($scope, categoryConfig, $http, $resource, showOptionsConfig, attributeOptionsConfig, HttpClientService) {
```
Comment (jkap) [inline app/productSearch/productSearch.js]: see above

File: app/recommendations/recommendations.js [hunk 3]
```diff
@@ -4,18 +4,11 @@ angular.module('bby-query-mixer.recommendations').controller('RecommendationsCtr
     '$scope',
     '$resource',
     'categoryConfig',
-    function ($scope, $resource, categoryConfig) {
+    'HttpClientService',
+    function ($scope, $resource, categoryConfig, HttpClientService) {
```
Comment (jkap) [inline app/recommendations/recommendations.js]: see above

**Comment by cmlizama** [comment]
@jkap:  took out the $resource dependencies

**Pull request merged**